### PR TITLE
chore: Remove unused code

### PR DESCRIPTION
**Pull request by brendongo**
PR checklist:

- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Changelog guidelines](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry)
- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/src/semgrep/core_runner.py [hunk 1]
```diff
@@ -565,7 +547,7 @@ def _plan_core_run(
 
         for i, rule in enumerate(rules):
             for language in rule.languages:
-                targets = self.get_files_for_language(language, rule, target_manager)
+                targets = list(target_manager.get_files_for_rule(language, rule.includes, rule.excludes, rule.id))
```
Comment (brendongo) [inline cli/src/semgrep/core_runner.py]:
```suggestion
                targets = list(
                    target_manager.get_files_for_rule(
                        language, rule.includes, rule.excludes, rule.id
                    )
                )
```

File: cli/src/semgrep/core_runner.py [hunk 2]
```diff
@@ -531,21 +528,6 @@ def _add_match_times(
...
             profiling_data.set_file_times(Path(t.path), rule_timings, t.run_time)
 
-    def get_files_for_language(
-        self, language: Language, rule: Rule, target_manager: TargetManager
-    ) -> List[Path]:
-        try:
-            targets = target_manager.get_files_for_rule(
-                language, rule.includes, rule.excludes, rule.id
-            )
-        except _UnknownLanguageError as ex:
```
Comment (brendongo) [inline cli/src/semgrep/core_runner.py:541]: get_files_for_rule never raised this exception. This exception is not raised anywhere in the codebase

File: cli/src/semdep/find_lockfiles.py [hunk 3]
```diff
@@ -28,92 +22,6 @@ class Node:
     val: Optional[Dict[Path, List[LockfileDependency]]]
 
 
-class DependencyTrie:
```
Comment (brendongo) [inline cli/src/semdep/find_lockfiles.py:31]: @mmcqd I think with the approach we took to get lockfiles we don't need this Trie anymore?

**Review by chmccreery — APPROVED** [review]
🙌

**Pull request merged**